app/(gcforms)/[locale]/(form administration)/form-builder/[id]/responses/[statusFilter]/components/ManageFormAccessDialog/ConfirmAction.tsx

```diff
@@ -0,0 +1,78 @@
+import { Button } from "@clientComponents/globals";
+import { useTranslation } from "@i18n/client";
+import { CancelIcon } from "@serverComponents/icons";
+import { useEffect, useRef, useState } from "react";
+
+type ConfirmActionProps = {
+  callback: () => Promise<boolean>;
+  icon?: React.ReactElement;
+  confirmString: string;
+  buttonLabel: string;
+  buttonTheme?: "primary" | "secondary" | "destructive" | "link";
+  children?: JSX.Element | string;
+};
+
+export const ConfirmAction = ({
+  callback,
+  icon = <CancelIcon />,
+  confirmString,
+  buttonLabel,
+  buttonTheme = "destructive",
+  children,
+}: ConfirmActionProps) => {
+  const { t } = useTranslation("manage-form-access");
+  const [confirm, setConfirm] = useState(false);
+  const [error, setError] = useState("");
+  const ref = useRef<HTMLDivElement>(null);
+
+  const handleOnClick = async () => {
```

Comment:
Do we need to handle ESC here as well?

Reply:
Tried capturing ESC to clear confirm, but the dialog is already listening for it and takes precedence.

Added an additional way of clearing when the Confirm button loses focus.